src/manifold/src/manifold.cpp

```diff
@@ -767,4 +830,104 @@ Manifold Manifold::TrimByPlane(glm::vec3 normal, float originOffset) const {
 }
 
 ExecutionParams& ManifoldParams() { return params; }
+
+Manifold Manifold::Hull(const std::vector<float>& vertProps, const int numProp,
+                        const double precision) {
+  const int len = vertProps.size() / numProp;
+  // FIXME: Should there be a hull specific error?
+  if (len < 4) return Invalid();
+  std::vector<glm::vec3> pts(len);
+  for (int i = 0; i < len; i++) {
+    const int j = i * numProp;
+    pts[i] = {vertProps[j], vertProps[j + 1], vertProps[j + 2]};
+  }
+  const auto trip = NonCollinearTriple(pts, precision);
+  const auto plane = PlaneOfPts(pts[trip.x], pts[trip.y], pts[trip.z]);
+  const auto [d, dIsAboveTrip] = NonCoplanar(pts, plane, precision);
+  if (d < 0) return Invalid();  // All points coplanar.
+  const int a = trip.x;
+  const int b = dIsAboveTrip ? trip.z : trip.y;
+  const int c = dIsAboveTrip ? trip.y : trip.z;
+  std::vector<glm::ivec3> triangles;
+  std::vector<glm::vec4> planes;
+  std::unordered_set<glm::ivec2, IVec2Hash> halfEdges{};
+  std::vector<int> dropped;
```

Comment:
We can avoid using dropped and kept vector by swapping dropped triangle/plane with the last element in the vector and then shrink the vector.

Reply:
The triangles to be dropped aren't necessarily the last one on the vector though.

Reply:
There can be many dropped triangles before new ones are added though. New drops would then swap the old drops back into the vector.

Reply:
Yeah true, I was thinking about parallelization and whether it is possible to do things using atomics. I guess this need some actual benchmark to see which way is faster.